csrc/apis/java/detector.cpp

```diff
@@ -0,0 +1,229 @@
+        };
+        static const unsigned char colors[19][3] = {
+            { 54,  67, 244},
+            { 99,  30, 233},
+            {176,  39, 156},
+            {183,  58, 103},
+            {181,  81,  63},
+            {243, 150,  33},
+            {244, 169,   3},
+            {212, 188,   0},
+            {136, 150,   0},
+            { 80, 175,  76},
+            { 74, 195, 139},
+            { 57, 220, 205},
+            { 59, 235, 255},
+            {  7, 193, 255},
+            {  0, 152, 255},
+            { 34,  87, 255},
+            { 72,  85, 121},
+            {158, 158, 158},
+            {139, 125,  96}
+        };
+
+        int color_index = 0;
+        for (int i = 0; i < *res_count; i++)
+        {
+            const mm_detect_t& det_result = bboxes[i];
+            // skip detections with invalid bbox size (bbox height or width < 1)
+            if ((det_result.bbox.right - det_result.bbox.left) < 1 || (det_result.bbox.bottom - det_result.bbox.top) < 1) {
+                continue;
+            }
+            // skip detections less than specified score threshold
+            if (det_result.score < 0.3) {
+                continue;
+            }
+            const unsigned char* color = colors[color_index % 19];
+            color_index++;
+
+            cv::Scalar cc(color[0], color[1], color[2]);
+            cv::rectangle(rgb, cv::Point{(int)det_result.bbox.left, (int)det_result.bbox.top},
+                cv::Point{(int)det_result.bbox.right, (int)det_result.bbox.bottom}, cc, 2);
+
+            char text[256];
+            sprintf(text, "%s %.1f%%", class_names[det_result.label_id], det_result.score * 100);
+
+            int baseLine = 0;
+            cv::Size label_size = cv::getTextSize(text, cv::FONT_HERSHEY_SIMPLEX, 0.5, 1, &baseLine);
+
+            int x = (int)det_result.bbox.left;
+            int y = (int)det_result.bbox.top - label_size.height - baseLine;
+            if (y < 0)
+                y = 0;
+            if (x + label_size.width > rgb.cols)
+                x = rgb.cols - label_size.width;
+
+            cv::rectangle(rgb, cv::Rect(cv::Point(x, y), cv::Size(label_size.width, label_size.height + baseLine)), cc, -1);
+
+            cv::Scalar textcc = (color[0] + color[1] + color[2] >= 381) ? cv::Scalar(0, 0, 0) : cv::Scalar(255, 255, 255);
+
+            cv::putText(rgb, text, cv::Point(x, y + label_size.height), cv::FONT_HERSHEY_SIMPLEX, 0.5, textcc, 1);
+        }
+        mmdeploy_detector_release_result(bboxes, res_count, 1);
+      }
+      draw_fps(jw, jh, rgb);
+      cvtColor(rgb, mat_image_src, cv::COLOR_RGB2RGBA, 4);
+      env->ReleaseIntArrayElements(jPixArr, cPixArr, 0);
+      return JNI_TRUE;
+    }
+    JNIEXPORT jboolean JNICALL loadModel(JNIEnv* env, jobject thiz, jobject assetManager, jstring work_dir, jint modelid)
+    {
+      AAssetManager* mgr = AAssetManager_fromJava(env, assetManager);
+      __android_log_print(ANDROID_LOG_DEBUG, "ncnn", "loadModel %p", mgr);
```

Comment:
@lzhangzz can `spdlog` support `__android_log_print`?

Reply:
there is `spdlog::android_logger_mt`, and `CreateDefaultLogger()` will try to use it when building for android, but it's not tested yet.